packages/widget/src/hooks/useFullPageInMaxHeightLayout.ts

```diff
@@ -0,0 +1,36 @@
+import { useTheme } from '@mui/material';
+import { useEffect } from 'react';
+import { defaultMaxHeight } from '../config/constants.js';
+import { getRelativeContainer } from '../utils/elements.js';
+import { useDefaultElementId } from './useDefaultElementId.js';
+
+// When the Widgets layout is set to use max height or in its default height mode the default behaviour
+// is for pages to use only the minimum height that they need to show their content.
+// On some pages you might want to always ensure that the page always up the maximum height available.
+// To do this just add this hook at the page level.
+// The other height layout will by default always occupy the maximum space available.
+// NOTE: this hook is implicitly tied to the widget height functionality in the
+//   AppExpandedContainer, RelativeContainer and CssBaselineContainer components as defined in AppContainer.ts
+//   CSS changes in those components can have implications for the functionality in this hook
```

Comment:
This thing becomes so complex 🥲

Reply:
I've decided to go with a way to let CSS carry the load here.

Have deleted this hook and created a full page container component

https://github.com/lifinance/widget/blob/ddbb8410419a178363aec8fc10f6a1248f9d73b1/packages/widget/src/components/FullPageContainer.tsx#L9-L14

The css for this lives in the AppContainer

https://github.com/lifinance/widget/blob/ddbb8410419a178363aec8fc10f6a1248f9d73b1/packages/widget/src/components/AppContainer.tsx#L93-L97

And just means on the pages that we swap out the PageContainer for the FullPageContainer

https://github.com/lifinance/widget/blob/ddbb8410419a178363aec8fc10f6a1248f9d73b1/packages/widget/src/pages/SelectChainPage/SelectChainPage.tsx#L69

This method was already being used for the SelectTokenPage by setting an id on the TokenList, I've just repurposed it here.

Reply:
In relation to the layout/height code I think there are still something that we can do to simplify things further in future.

- Consolidate the logic and styles in the AppContainer.tsx files AppExpandedContainer, RelativeContainer and CssBaselineContainerProps. It does get confusing where the height should be set to get the desired result for layouts
- useTokenListHeight hook could potentially be replaced with a css implementation - I've managed to create a list in a side project that  resizes in the way that the token list in our app does using just css. Though I have had a failed attempt at getting this working already I do think that at some point its worth another attempt.